### PR TITLE
Tor disabled cleanup

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/14518

This PR cleans up tor/data and tor/watch used by Tor when Tor is disabled. Also prompt users to do disable and then re-enabled when Tor status stays disconnected

Screen Shot 2021-05-05 at 10 06 12

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Initialization stuck
1. @fmarier can share with QA a user-supplied broken profile for Mac that reproduces this problem without having to disable one's Internet connection.
2. Copy the `tor/data` from step 1 to `user_data_dir` to replace the original one
3. Open Tor window should experience initialization stuck at some point and also see the tip message
4. Close Tor window and go to `brave://settings/extensions` and disable Tor and re-enable it again
5.  Open Tor window
6. Tor status should be connected after initiating complete

### Prompt
1. Turn off internet access (for simulating constant disconnected state of Tor)
2. Open Tor window
3. We will see Tor status is disconnected and there is a tip prompting users to disable and re-enable Tor
4. Click on the link
5. It should lead us to brave://settings/extensions
6. Turn internet back on and go back to Tor window
7. When the status is connected, the tip should go away

### Disable Tor
1. Open a Tor window and make sure `user_data_dir` has a `tor/data` and `tor/watch`
2. Go to `brave://settings/extensions` and disable Tor
3. `tor` folder in `user_data_dir` should be empty

### CODE REVIEW

File: components/resources/brave_components_strings.grd [hunk 1]
```diff
@@ -334,6 +334,7 @@
       <message name="IDS_BRAVE_PRIVATE_NEW_TAB_TOR_STATUS_CONNECTED" desc="">Connected</message>
       <message name="IDS_BRAVE_PRIVATE_NEW_TAB_TOR_STATUS_DISCONNECTED" desc="">Disconnected</message>
       <message name="IDS_BRAVE_PRIVATE_NEW_TAB_TOR_STATUS_INITIALIZING" desc="">Initializing [[percentage]]%</message>
+      <message name="IDS_BRAVE_PRIVATE_NEW_TAB_TOR_TIP" desc="">Having problems with Tor? Try <ph name="BEGIN_LINK">$1</ph>disabling private window with Tor<ph name="END_LINK">$2</ph> and turning it back on again</message>
```
Comment (fmarier) [inline components/resources/brave_components_strings.grd:337]: There's a missing period at the end of the sentence.

**Review by fmarier — COMMENTED** [review]
Do you know of a way to break the `tor/data` directory so that QA could verify that the reset is working?

**Comment by darkdh** [comment]
> Do you know of a way to break the tor/data directory so that QA could verify that the reset is working?
yes, need the tor/data from your friend.

File: components/resources/brave_components_strings.grd (hunk 1 above)
Comment (darkdh) [reply components/resources/brave_components_strings.grd]: fixed

**Review by fmarier — APPROVED** [review]
New message looks good.

**Review by simonhong — APPROVED** [review]
++

**Pull request merged**